operator/v1/types_dns.go

```diff
@@ -376,6 +409,20 @@ const (
 	NetworkResolverType   UpstreamType = "Network"
 )
 
+// ProtocolStrategy is a preference for the protocol to use for DNS queries.
+// +kubebuilder:validation:Enum:=TCP;""
```

Comment:
Sweet. I did not know one can apply the validation directly to a type and have it distributed into both `ForwardPlugin` and `UpstreamResolvers` structs.